rust/toolchain.bzl

```diff
@@ -268,6 +268,11 @@ def _rust_toolchain_impl(ctx):
         linking_context = linking_context,
     )
 
+    rust_std_files = depset(transitive = [
+        rust_std[DefaultInfo].files,
+        rust_std[DefaultInfo].default_runfiles.files,
```

Comment:
Why this change with `default_runfiles`? This is surprising. It's also surprising that it seems `rust_std` has some runfiles that are not in `files`, and here we want to make them available at build time. I'm really curious to know the story now :)

Reply:
I have no story around `rust_std` using runfiles but had the thought when I wrote `DefaultInfo` and leaned toward the less breaking change. I expect most users to be using what's defined in `rust_register_toolchains` which would not have runfiles. I can remove this if you think it's better 😄

Reply:
> leaned toward the less breaking change

But the original code didn't check runfiles, so it's strictly less breaking change to omit that line from this PR. If you think this is something that needs to be fixed, I'd recommend a separate PR, justification, and a test if possible..